src/OpenApi/SchemaDefinition/Entity/EntityAdapter.php

```diff
@@ -148,10 +150,23 @@ protected function getProperties(string $type): array
 
 	private function getPropertyType(ReflectionProperty $property): ?string
 	{
-		//TODO - fix typed properties support
-		//if (PHP_VERSION_ID >= 70400 && ($type = Reflection::getPropertyType($property) !== null)) {
-		//	return ($property->getType()->allowsNull() ? '?' : '') . $type;
-		//}
+		if (PHP_VERSION_ID >= 70400 && ($type = Type::fromReflection($property)) !== null) {
+			if ($type->isSingle() && count($type->getNames()) === 1) {
+				return $type->getNames()[0];
+			}
+
+			if ($type->isUnion()
+				|| ($type->isSingle() && count($type->getNames()) === 2) // nullable type is single but returns name of type and null in names
+			) {
+				return implode('|', $type->getNames());
+			}
+
+			if ($type->isIntersection()) {
+				return implode('&', $type->getNames());
+			}
+
+			throw new RuntimeException(sprintf('Could not parse type "%s"', $property));
```

Comment:
I'm not sure if we should throw an exception here, or we should quietly fall back to annotations here (e.g. when PHP starts supporting mixing unions and intersections in types)

Reply:
I think it's OK this way. :-)